meta-mender-core/recipes-bsp/grub/patches/0001-workaround-wrong-detection-headeris-in-warrior.patch

```diff
@@ -0,0 +1,76 @@
+From 469bf02510cda9d7bb3d3f3521a221845054786a Mon Sep 17 00:00:00 2001
+From: Lluis Campos <lluis.campos@northern.tech>
+Date: Fri, 13 Sep 2019 17:10:43 +0200
+Subject: [PATCH] Workaround wrong detection headeris in warrior
+
+For some uknown reason, in Yocto warrior the grub configure script is
+not able to find two header files: wchar.h and sysmacros.h, which
+triggers the build to abort with compile errors.
+
+The header files are actually there for glibc and probably most of libc
+versions used for native compile. As we are interested in grub-editenv,
+and not compiling grub itself, it should be safe to assume tha the
+headers are present in the sysroot for the toolchain.
+
+Signed-off-by: Lluis Campos <lluis.campos@northern.tech>
+---
+ grub-core/gnulib/wchar.in.h     | 5 +++--
+ grub-core/osdep/linux/getroot.c | 7 +++----
+ grub-core/osdep/unix/getroot.c  | 7 +++----
+ 3 files changed, 9 insertions(+), 10 deletions(-)
+
+diff --git a/grub-core/gnulib/wchar.in.h b/grub-core/gnulib/wchar.in.h
+index b6e4362..6de02fd 100644
+--- a/grub-core/gnulib/wchar.in.h
++++ b/grub-core/gnulib/wchar.in.h
+@@ -74,9 +74,10 @@
+ /* Include the original <wchar.h> if it exists.
+    Some builds of uClibc lack it.  */
+ /* The include_next requires a split double-inclusion guard.  */
+-#if @HAVE_WCHAR_H@
++/* Assume the header file exists. For some strange reason,
++ * configure resolves the header to no exist when building
++ * on Yocto warrior branch */
+ # @INCLUDE_NEXT@ @NEXT_WCHAR_H@
+-#endif
```

Comment:
Is it difficult to fix `configure.ac` instead? I assume that such a patch is much more likely to be accepted upstream.

Reply:
I tried hard! I really couldn't figure it out, specially the second one, the header `sysmacros.h`. For this first one, I located where the corresponding code in autotools live so I could hack it there, but finally just for being consistently hacky I decided to patch the source code.

ATM I am dealing with other issues (tests failing and so on) so I will keep this workaround for now and give it a second go at the end of my task.

Reply:
Lol, "consistently hacky"! Sounds like it's more difficult than it's worth then.